### ISSUE TITLE
Fable doesn't show error in watch mode with Vite when changing name of module function

### ISSUE THREAD

**Issue by alfonsogarciacaro** [fable-compiler/Fable#3184]
Transferred from https://github.com/alfonsogarciacaro/Feliz.Solid/issues/1

@kunjee17 wrote:

Here is how to reproduce the issue.

Go to the project and run the project as it it should run. Once project is running go to component.fs file and change the name of any function / component. Desired behavior should be, compilation error on terminal. But here terminal is very much clean without any issue. and without any issue and also project will not work. You can actually see the error in browser console for jsx file.

@dstien wrote

Do the compilation output appear in your terminal if you add clearScreen: false in vite.config.js?

Like this:

```js
import { defineConfig } from 'vite';
import solidPlugin from 'vite-plugin-solid';

export default defineConfig({
  clearScreen: false,
  plugins: [solidPlugin()],
});
```

@kunjee17 wrote:

It does show file which it compiles but no error. Fable successfully compiles the file ( fs to jsx ) and hand it over to dev server. But as there is error in jsx file there will be no hmr and also error in console. Desire behavior should be error in terminal .

**Comment by alfonsogarciacaro** [comment]
Thanks for reporting @kunjee17 and for your help @dstien

I think I know what the problem is. In watch mode Fable only compiles files with modifications. So in this case the error when changing the function name is not in Components.fs but in App.fs. Fable does have some rules to build a dependency tree, but in the case of module functions I think so far we've just relied on Webpack throwing an error. This is not the case with Vite as it only recompiles the edited file. We may need to extend the dependency tree so it covers any file that imports another.
